api/models/spell.py

```diff
@@ -0,0 +1,229 @@
+        """Logic to import from the v1 file spec."""
+        self.name = json["name"]
+        self.slug = slugify(json["name"])
+        if "desc" in json:
+            self.desc = json["desc"]
+        if "higher_level" in json:
+            self.higher_level = json["higher_level"]
+        if "page" in json:
+            self.page = json["page"]
+        # Logic to set an integer based on v1 file import spec (a string).
+        if "range" in json:
+            # If the spell range includes an effect description like "(10 ft
+            # cube)", remove.
+            range_trimmed = str(json['range'].split(" (")[0]).lower()
+
+            # Set target range sort based on the following rules:
+            # Self = 0
+            # Touch = 1
+            # Any other distance is recorded in feet.
+            # "Sight" is recorded as 9999
+            # Unlimited (same-plane) is recorded as 99990
+            # Unlimited (multi-plane) is recorded as 99999
+
+            if 'self' in range_trimmed:
+                # Various spells in deep magic include effect shape.
+                self.target_range_sort = 0
+            if range_trimmed == 'touch':
+                self.target_range_sort = 1
+
+            if range_trimmed == 'sight':
+                self.target_range_sort = 9999
+
+            if range_trimmed == 'unlimited':
+                self.target_range_sort = 99999
+
+            if range_trimmed == 'special':
+                # For now, assume it's single plane
+                self.target_range_sort = 99990
+
+            if range_trimmed.endswith(
+                    'miles') or range_trimmed.endswith('mile'):
+                # Assume it's in the format "xx Miles"
+                miles = int(json['range'].split(" ")[0])
+                self.target_range_sort = miles * 5280
+
+            if range_trimmed.endswith('feet') or range_trimmed.endswith('ft.'):
+                # Assume it's in the format "xx Miles"
+                feet = int(range_trimmed.split(" ")[0])
+                self.target_range_sort = feet
+
+            self.range = json["range"]
+        if "components" in json:
+            # Set defaults to False, and then set to true based on string of
+            # "V, S, M"
+            self.requires_verbal_components = False
+            self.requires_somatic_components = False
+            self.requires_material_components = False
+            if 'v' in json['components'].lower():
+                self.requires_verbal_components = True
+            if 's' in json['components'].lower():
+                self.requires_somatic_components = True
+            if 'm' in json['components'].lower():
+                self.requires_material_components = True
+
+        if "material" in json:
+            self.material = json["material"]
+
+        # Logic to set boolean based on v1 file import spec (a string).
+        self.can_be_cast_as_ritual = False
+        if "ritual" in json:
```

Comment:
nit: L187-193 can be compressed:

```py
self.can_be_cast_as_ritual = json.get("ritual", None) == "yes"
```

Reply:
I have adjusted all of the imports to be simpler, but I didnt' quite one-liner them. Some of them I'm parsing text, so two lines feels more natural to me, but It's cleaner than it was based on your input.